Content/RootScript/BaseClasses/Utility/AngelMath_vector.as

```diff
@@ -0,0 +1,257 @@
+
+shared class Vector2d { // : Vector2f
```

Comment:
I would recommend renaming these/putting them in a namespace that indicates a more limited scope.

Vectors generally should not implement `operator*` as there are multiple products between vectors, so the operator is inadequate to represent any single one of them.

What I would recommend is calling these `UIVector` or `UIPoint`, don't expose dot or cross product and overload `operator*` to be per-element multiplication.

Reply:
Generally, Vector*Vector should always point to multiply (x1*x2, y1*y2) unless otherwise specified. this is the most common operation. same for division. the others like dot and cross can simply be vector.dot(other) and vector.cross(other). This allows chaining vector math together, which will be very useful in managing world objects.

Reply:
> Generally, `Vector*Vector` should always point to multiply `(x1*x2, y1*y2)` unless otherwise specified. this is the most common operation.
Being common does not mean that it is appropriate for the design considerations for this project. PGE's vectors will not implement `operator*` in this way as its meaning is potentially unclear due to the existence of the other vector products. What I'm suggesting here is utilizing specialized types where you would be allowed to implement `operator*` by eliminating other operations that could lead to confusion, indicating purpose through the name of the type and preventing conversions to types used in other contexts where this operation is not the only one applicable.

In contexts where we may want to do dot or cross product (which will likely be a common case when working with world-space vectors), I would much rather write `vec1.perElementProduct(vec2)` over `vec1 * vec2` when I actually do want to perform per-element product. Intent is simply a lot clearer in the former.

Reply:
For the moment, I have reverted the engine so these functions aren't actually used anywhere. The overloads can be removed for the time being without issue, and we'll deal with world-space vectors when we get to it, as that is my concern as well.

Reply:
This is resolved as per specific class instances on an as-needed bases, i.e namespace::Vector2d;